### PR TITLE
Rename TypeName.nullable into TypeName.isNullable

### PR DESCRIPTION
**Pull request by Egorand**

### CODE REVIEW

File: src/main/java/com/squareup/kotlinpoet/TypeName.kt [hunk 1]
```diff
@@ -412,7 +412,7 @@ class LambdaTypeName private constructor(
   }
 
   fun copy(
-    nullable: Boolean = this.nullable,
+    nullable: Boolean = this.isNullable,
```
Comment (shaishavgandhi05) [inline src/main/java/com/squareup/kotlinpoet/TypeName.kt]: Should the `copy` parameter also be renamed to `isNullable` for consistency?

Comment (Egorand) [reply src/main/java/com/squareup/kotlinpoet/TypeName.kt]: I think `isNullable` reads better when it's an accessor ("is this TypeName nullable?"), and `nullable` reads better when it's a parameter ("I'm creating a nullable TypeName").

Comment (shaishavgandhi05) [reply src/main/java/com/squareup/kotlinpoet/TypeName.kt]: Hmm but then the constructor parameter is `isNullable` 🙂

Comment (Egorand) [reply src/main/java/com/squareup/kotlinpoet/TypeName.kt]: `TypeName` is abstract and its subtypes don't have public constructors

Comment (shaishavgandhi05) [reply src/main/java/com/squareup/kotlinpoet/TypeName.kt]: Oops missed that!

**Pull request merged**